analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/RuleChecks/EmptyCollectionsShouldNotBeEnumeratedBase.cs

```diff
@@ -138,6 +138,11 @@ protected override ProgramState PreProcessSimple(SymbolicContext context)
         return state;
     }
 
+    private static bool IsEmptyObject(IOperation operation) =>
+        operation.AsObjectCreation() is { } objectCreation
+        && objectCreation.Type.IsAny(TrackedCollectionTypes)
+        && objectCreation.Constructor.Parameters.All(x => !x.Type.DerivesOrImplements(KnownType.System_Collections_IEnumerable));
```

Comment:
This could propagate Empty constraint when present on the 1st argument (of correct type)